feat: enhance compliant form

AvivAbachi opened the pull request:
[list of complient (he)](https://docs.google.com/spreadsheets/d/1tuiG33ikUYYlzekKe3o0WLnRotxfxq6UCmP2fPMKxlI/edit?usp=sharing)

Server PR [#20](https://github.com/hasadna/open-bus-backend/pull/20)

- [x] Client: Build a modular Form
- [x]  Client: Fix Stories
- [x] Server: Update the schema  
- [x] Server: Implement the TemplateBuild for the complaint base type  
- [x] Package: Update the client API to reflect the new schema  
- [x] Client: Update the client API + Integrated  To API
- [x] Server: Fix XML
- [x] Client: Error Handeling
- [x] Client: Fix submit data

AvivAbachi committed:
feat: init

AvivAbachi committed:
integrate to api

AvivAbachi committed:
update complaint form

AvivAbachi committed:
clean no need files

AvivAbachi committed:
revest changes

AvivAbachi committed:
fix: update backend API base path and refactor complaint form submission

AvivAbachi committed:
Merge branch 'main' into feat/complent-form

NoamGaash commented on src/api/apiConfig.ts:
```diff
@@ -18,7 +18,7 @@ export const GTFS_API = new GtfsApi(STRIDE_API_CONFIG)
 export const USER_CASE_API = new UserCasesApi(STRIDE_API_CONFIG)
 
-const BACKEND_API_BASE_PATH = process.env.VITE_BACKEND_API
+const BACKEND_API_BASE_PATH = 'Http://127.0.0.1:3001' //process.env.VITE_BACKEND_API
```
const BACKEND_API_BASE_PATH = process.env.VITE_BACKEND_API ?? 'http://127.0.0.1:3001'

NoamGaash commented on src/pages/components/map-related/MapLayers/ComplaintModal.tsx:
```diff
@@ -54,156 +57,209 @@ const complaintTypes: complaintType[] = [
   { value: 'station_signs', label: 'station_signs' },
 ]
 
+function IDValidator(id?: string) {
+  if (id === undefined || id === '' || id.length !== 9) return true
+  const num = Number(id)
+  if (!num || isNaN(num) || num <= 0) return false
+  let sum = 0
+  for (let i = 0; i < id.length; i++) {
+    const n = Number(id[i]) * ((i % 2) + 1)
+    sum += Math.floor(n / 10) + (n % 10)
+  }
+  return sum % 10 === 0
+}
+
```
cool!
can be placed on another file, but it's also fine to keep it here

NoamGaash commented on src/pages/components/map-related/MapLayers/ComplaintModal.tsx:
```diff
@@ -54,156 +57,209 @@ const complaintTypes: complaintType[] = [
       ) : (
         <Dialog
-          dir={textDirection}
+          dir={i18n.dir()}
```
better!

NoamGaash reviewed (commented):
So far it looks great!

AvivAbachi committed:
refactor: complaint form with dynamic fields and config-driven approach

AvivAbachi committed:
refactor: move complaint form to config‑driven approach

The modal now imports field definitions, mappings, and rendering logic from `ComplaintModalConfig.tsx`.
Inline `renderField` implementation and type declarations have been removed, simplifying maintenance and adding support for new train complaint types.

AvivAbachi committed:
refactor

AvivAbachi committed:
refactor(config): update complaint modal with collapse and select fields

AvivAbachi committed:
refactor(complaint-modal): update API integration and dynamic field handling

AvivAbachi committed:
refacto: improve field handling and validation logic

AvivAbachi committed:
refactor(complaint-modal): restructure complaint modal components and consolidate field handling

AvivAbachi committed:
feat: enhenc date picker and BusOpreatorSelect

AvivAbachi committed:
fix: ride data

AvivAbachi committed:
refactor: consolidate query hooks for complaint form

AvivAbachi committed:
refactor: improve complaint modal route handling

AvivAbachi committed:
refacto: simplify boarding station query and rename direction fields

AvivAbachi committed:
fix: adjust complaint modal fields and queries

AvivAbachi committed:
refactor: simplify field handling and dynamic rendering

AvivAbachi committed:
refactor: refactor data fetching and form initialization

AvivAbachi committed:
refactor: improve form queries and dynamic field rendering

AvivAbachi committed:
perf: optimize data fetching and caching for form queries

AvivAbachi committed:
feat: add route data to complaint modal story

AvivAbachi committed:
refactor: refine query keys and form field dependencies

AvivAbachi committed:
feat: auto-select route based on line data

AvivAbachi committed:
feat(complaint-modal): persist user details in complaint form

AvivAbachi committed:
refactor: enhance form handling and type definitions

AvivAbachi committed:
feat: add comprehensive fields and localize

AvivAbachi committed:
refactor: clarify bus operator and query variable names

AvivAbachi committed:
feat: add comprehensive form validation

AvivAbachi committed:
feat: enhance form with new fields and mappings

AvivAbachi committed:
feat: enhance complaint type mappings

AvivAbachi committed:
feat: refine add frequency

AvivAbachi committed:
feat: add station management fields to complaint form

AvivAbachi committed:
feat(complaint-form): implement station management and new fields

AvivAbachi committed:
feat(complaint-form): refine station add/remove logic and fields

AvivAbachi committed:
refactor: refine complaint form labels and messages

AvivAbachi committed:
feat(complaint-form): enhance line switch complaint guidance

AvivAbachi committed:
refactor: consolidate and simplify validation rule access

AvivAbachi committed:
Merge branch 'main' into feat/complent-form

NoamGaash commented on src/locale/he.json:
```diff
@@ -218,13 +218,14 @@
+  "add_station": "הוסף תחנה",
+  "remove_station": "הסר תחנה",
   "willing_to_testify_mot": " הנני מאשר/ת כי אהיה מוכן/ה למסור עדות במשרד התחבורה",
   "willing_to_testify_court": "הנני מאשר/ת כי אהיה מוכן/ה למסור עדות בבית המשפט",
```
Those fields are from the MOT form? :exploding_head:

AvivAbachi committed:
Merge branch 'main' into feat/complent-form

AvivAbachi committed:
test: update eyes wait strategy for complaint modal

AvivAbachi committed:
refactor: rename form fields and locale keys

AvivAbachi committed:
refactor(complaint-modal): clarify bus operator field

AvivAbachi committed:
Merge branch 'main' into feat/complent-form

AvivAbachi committed:
feat(complaint-form): add new fields and dynamic logic

AvivAbachi committed:
refactor(complaint-modal): clarify findIndex logic

AvivAbachi committed:
feat(complaint-form): add city field

AvivAbachi committed:
complain well be only in bus

AvivAbachi committed:
mobile validation

AvivAbachi committed:
feat(complaint-form): add new fields and improve form inputs

AvivAbachi committed:
fix(complaint-form): refine mobile number validation regex

AvivAbachi committed:
Merge branch 'main' into feat/complent-form

AvivAbachi committed:
refactor

AvivAbachi committed:
imporve handel submit

AvivAbachi committed:
feat(complaint-form): integrate complaints API client and refactor form submission

AvivAbachi committed:
Merge branch 'main' into feat/complent-form

AvivAbachi committed:
refactor(api): centralize api client configuration

AvivAbachi committed:
add debug mode

AvivAbachi committed:
Merge branch 'main' into feat/complent-form

AvivAbachi committed:
Merge branch 'main' into feat/complent-form

AvivAbachi committed:
feat: add compline title

AvivAbachi committed:
Merge branch 'main' into feat/complent-form

AvivAbachi committed:
fix: update property names in ComplaintModal for consistency

AvivAbachi committed:
feat: add error alert to ComplaintModal and conditionally render applyContent field

AvivAbachi committed:
fix: date

AvivAbachi committed:
refactor: comment out unused fields and queries in ComplaintModal and related files

AvivAbachi committed:
feat: enhance EasterEgg component in ComplaintModal to support autohide and debug state

AvivAbachi committed:
remove log

AvivAbachi committed:
improve complaint type  on debug

AvivAbachi committed:
Merge branch 'main' into feat/complent-form

AvivAbachi committed:
Merge branch 'main' into feat/complent-form

NoamGaash reviewed (commented):
Observations -
1. you have removed the "complaint" easter egg. Is it on purpose? Are we sure that we want to expose this button to everyone? Maybe we can do some more tests before exposing it to the public. We can create a google form and ask friends and colleagues to help us with real data to perform some more e2e manual tests with against the MOT production.
2. I tried to do the following:
   - made sure that the `debug` flag is activated
   - sent a test complaint
   - copied the `xml` from the response payload of the `send` endpoint
   - tried to use `יבא` in the actual form.
 I can see the following error, which can indicate that we might have date formatting issue:
```
Uncaught TypeError: Cannot read properties of null (reading 'getFullYear')
    at FormScripts_combined.js:2106:10422
    at a (FormScripts_combined.js:62:19000)
    at s.Ub.ft [as La] (FormScripts_combined.js:62:18752)
    at t.notifySubscribers (FormScripts_combined.js:62:13382)
    at t.W (FormScripts_combined.js:62:15227)
    at t (FormScripts_combined.js:62:15116)
    at l (FormScripts_combined.js:69:5424)
    at FormScripts_combined.js:69:4891
    at g (FormScripts_combined.js:69:6122)
    at l (FormScripts_combined.js:69:4584)
```
but despite a very broken layout, seems like we do capture most of the essential fields:

<img width="2366" height="16384" alt="Image" src="https://github.com/user-attachments/assets/45a05c74-2126-4c13-890b-b80f996bdd6b" />

AvivAbachi committed:
fix typo error

AvivAbachi committed:
Merge branches 'feat/complent-form' and 'feat/complent-form' of https://github.com/hasadna/open-bus-map-search into feat/complent-form

AvivAbachi committed:
reorder types

AvivAbachi committed:
fix button map css

AvivAbachi committed:
complaints translation warp

AvivAbachi committed:
fix map css

AvivAbachi committed:
fix types

AvivAbachi committed:
fix import

AvivAbachi committed:
Merge branch 'main' into feat/complent-form

AvivAbachi committed:
Merge branch 'main' into feat/complent-form

AvivAbachi committed:
Merge branch 'main' into feat/complent-form

AvivAbachi committed:
Merge branch 'main' into feat/complent-form

AvivAbachi committed:
Merge remote-tracking branch 'origin/main' into feat/complent-form

AvivAbachi committed:
update

AvivAbachi committed:
feat: add mock handlers for operators and cities in ComplaintModal stories

AvivAbachi committed:
Merge branch 'main' into feat/complent-form
